# Description: Reset the state of g_exited_main_message_loop when restarting the message
#  loop, as it is checked in the constructor for RenderProcessHostImpl
# Author: Chris Coulson <chris.coulson@canonical.com>

diff --git a/content/browser/browser_main_runner.cc b/content/browser/browser_main_runner.cc
--- a/content/browser/browser_main_runner.cc
+++ b/content/browser/browser_main_runner.cc
@@ -114,16 +114,17 @@ class BrowserMainRunnerImpl : public Bro
 
     // Return -1 to indicate no early termination.
     return -1;
   }
 
   virtual int Run() OVERRIDE {
     DCHECK(initialization_started_);
     DCHECK(!is_shutdown_);
+    g_exited_main_message_loop = false;
     main_loop_->RunMainMessageLoopParts();
     return main_loop_->GetResultCode();
   }
 
   virtual void Shutdown() OVERRIDE {
     DCHECK(initialization_started_);
     DCHECK(!is_shutdown_);
 #ifdef LEAK_SANITIZER
